sinks/sinks_service_test.go

```diff
@@ -118,18 +118,49 @@ func TestUpdateSink(t *testing.T) {
 
 	noConfig := sk
 	noConfig.Config = make(map[string]interface{})
-	noConfig.Name, _ = types.NewIdentifier("noConfig")
 
-	noTags := sk
+	noTagName, _ := types.NewIdentifier("noTagSink")
+	noTags := sinks.Sink{
+		Name:        noTagName,
+		Description: "An example prometheus sink",
+		Backend:     "prometheus",
+		State:       sinks.Unknown,
+		Error:       "",
+		Config:      map[string]interface{}{"remote_host": "data", "username": "dbuser"},
+		Tags:        map[string]string{"cloud": "aws"},
+	}
+	noTags, err = service.CreateSink(context.Background(), token, noTags)
+	require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+	noTags.Backend = ""
 	noTags.Tags = make(map[string]string)
-	noTags.Name, _ = types.NewIdentifier("noTags")
 
 	noDescription := sk
-	noDescription.Tags = make(map[string]string)
-	noDescription.Name, _ = types.NewIdentifier("noDesc")
+	noDescription.Description = ""
+
+	newTagsSink := sk
+	newTags := types.Tags{"cloud": "aws", "test": "true"}
+	newTagsSink.Tags = types.Tags{"test": "true"}
+
+	newVTagName, _ := types.NewIdentifier("newKTagSink")
+	newVTagSink := sinks.Sink{
+		Name:        newVTagName,
+		Description: "An example prometheus sink",
+		Backend:     "prometheus",
+		State:       sinks.Unknown,
+		Error:       "",
+		Config:      map[string]interface{}{"remote_host": "data", "username": "dbuser"},
+		Tags:        map[string]string{"cloud": "aws"},
+	}
+	newVTagSink, err = service.CreateSink(context.Background(), token, newVTagSink)
+	require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+	newVTagSink.Backend = ""
+
+	newVTagSink.Tags = map[string]string{"cloud": "true"}
+	newVTag := types.Tags{"cloud": "true"}
 
 	cases := map[string]struct {
 		sink  sinks.Sink
+		tags  types.Tags
 		token string
 		err   error
 	}{
```

Comment:
Are you using the tags, just for validation of tags?
If so, I would use something like this:
```suggestion
    cases := map[string]struct {
         incomingSink sinks.Sink
         expectedTags types.Tags
         token string
         err error
         }{
```

Reply:
yes, I'm using it for that, I'll use your suggestion